parsons/zoom/zoom.py

```diff
@@ -23,15 +26,49 @@ class Zoom:
             variable set.
     """
 
-    def __init__(self, api_key=None, api_secret=None):
+    def __init__(self, account_id=None, client_id=None, client_secret=None):
+        self.account_id = check_env.check("ZOOM_ACCOUNT_ID", account_id)
+        self.client_id = check_env.check("ZOOM_CLIENT_ID", client_id)
+        self.__client_secret = check_env.check("ZOOM_CLIENT_SECRET", client_secret)
+
+        self.client = APIConnector(uri=ZOOM_URI)
+
+        access_token = self.__generate_access_token()
 
-        self.api_key = check_env.check("ZOOM_API_KEY", api_key)
-        self.api_secret = check_env.check("ZOOM_API_SECRET", api_secret)
-        self.client = APIConnector(ZOOM_URI)
+        self.client.headers = {
+            "Authorization": f"Bearer {access_token}",
+            "Content-type": "application/json",
+        }
+
+    def __generate_access_token(self) -> str:
+        """
+        Uses Zoom's OAuth callback URL to generate an access token to query the Zoom API
+
+        `Returns`:
+            String representation of access token
+        """
 
-    def refresh_header_token(self):
-        # Generate a token that is valid for 30 seconds and update header. Full documentation
-        # on JWT generation using Zoom API: https://marketplace.zoom.us/docs/guides/auth/jwt
+        temp_client = APIConnector(
+            uri=ZOOM_URI, auth=(self.client_id, self.__client_secret)
+        )
```

Comment:
I assume the reason for the second client is that this one needs the `auth` param passeed in and the other one needs the bearer token?

Reply:
yep exactly, happy to workshop this if you think it's a little clunky